File: Assets/Code/Outfit7/Raven/TrackComponents/Events/Common/RavenCallFunctionEvent.cc
#ifdef STARLITE
#include "RavenCallFunctionEvent.h"
#include "RavenCallFunctionEvent.cs"

#include <RavenSequence.h>

namespace Starlite {
    namespace Raven {
        RavenCallFunctionEvent::RavenCallFunctionEvent() {
        }

        bool RavenCallFunctionEvent::IsBarrier() const {
            return m_IsBarrier && m_TrackIndex == 0 && m_SubTrackIndex == 0;
        }

        bool RavenCallFunctionEvent::IsValid() const {
            if (!RavenTriggerEvent::IsValid()) {
                return false;
            }

            if (!m_Property) {
                return false;
            }

            return m_Property->IsValid();
        }

        const RavenAnimationDataComponentBase* RavenCallFunctionEvent::GetAnimationDataEditorOnly() {
            return nullptr;
        }

        const RavenPropertyComponent* RavenCallFunctionEvent::GetPropertyComponent() const {
            return m_Property;
        }

        const RavenTriggerPropertyComponentBase* RavenCallFunctionEvent::GetProperty() const {
            return m_Property;
        }

        void RavenCallFunctionEvent::DestroyEditor(Ptr<RavenSequence> sequence) {
            RavenTriggerEvent::DestroyEditor(sequence);
            if (m_Property) {
                m_Property->DestroyEditor(sequence);
            }
        }

        void RavenCallFunctionEvent::Initialize(Ptr<RavenSequence> sequence) {
            RavenTriggerEvent::Initialize(sequence);
            m_Property->Initialize(sequence);
        }

        void RavenCallFunctionEvent::SetProperty(Ref<RavenTriggerPropertyComponentBase>& value) {
            m_Property = value;
        }

        void RavenCallFunctionEvent::OnEnterCallback(int frame) {
            m_Property->OnEnter();
        }

        void RavenCallFunctionEvent::OnSetTargetEditor(Ptr<RavenSequence> sequence, Ref<SceneObject>& target) {
            if (m_Property) {
                m_Property->SetTarget(target);
                m_Property->Initialize(sequence);
            }
        }
    } // namespace Raven
} // namespace Starlite
#endif

File: Assets/Code/Outfit7/Raven/TrackComponents/Events/Common/RavenCallFunctionEvent.cs
//{ "engine_register_objects":["Starlite__Raven__RavenCallFunctionEvent"] }
#if !__cplusplus
#pragma warning disable
namespace Starlite {

    namespace Raven {
    
        // RavenCallFunctionEvent
        [Starlite.ObjectClassAttribute("Starlite::Raven::RavenCallFunctionEvent")]
        public sealed partial class RavenCallFunctionEvent : global::Starlite.Raven.RavenTriggerEvent {
        	// Properties
        	[UnityEngine.SerializeField] [Raven.VisibleConditionAttribute("m_TrackIndex == 0 && m_SubTrackIndex == 0")] [UnityEngine.HeaderAttribute("Settings")] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] private bool m_IsBarrier;
        	[UnityEngine.SerializeField] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] private global::Starlite.Raven.RavenTriggerPropertyComponentBase m_Property;
        }
        
    }
    
}
#pragma warning restore
#else
// RavenCallFunctionEvent
namespace Starlite { namespace Raven { 
SIMPLEMENT_CLASS_WITH_PROPERTIES_IN_CLASS_SCOPED(Starlite::Raven::RavenCallFunctionEvent, RavenCallFunctionEvent, RavenCallFunctionEvent, Starlite::Raven::RavenTriggerEvent, Starlite::SceneObjectComponent, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenCallFunctionEvent, m_IsBarrier, m_IsBarrier, kBool, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenCallFunctionEvent, m_Property, m_Property, kObject, kUnknown, Starlite::Raven::RavenTriggerPropertyComponentBase::TypeId, "")
SIMPLEMENT_CLASS_END(RavenCallFunctionEvent)
} } 
SIMPLEMENT_CLASS_SCOPE(Starlite__Raven__RavenCallFunctionEvent, Starlite::Raven, RavenCallFunctionEvent)
#endif
